src/libs/actions/Session.js

```diff
@@ -33,37 +32,102 @@ function setSuccessfulSignInData(data, exitTo) {
 }
 
 /**
- * Sign in with the API
+ * Create an account for the user logging in.
+ * This will send them a notification with a link to click on to validate the account and set a password
  *
- * @param {String} partnerUserID
- * @param {String} partnerUserSecret
+ * @param {String} login
+ */
+function createAccount(login) {
+    Onyx.merge(ONYXKEYS.SESSION, {error: ''});
+
+    API.User_SignUp({
+        email: login,
+    }).then((response) => {
+        if (response.jsonCode !== 200) {
+            let errorMessage = response.message || `Unknown API Error: ${response.jsonCode}`;
+            if (!response.message && response.jsonCode === 405) {
+                errorMessage = 'Cannot create an account that is under a controlled domain';
+            }
+            Onyx.merge(ONYXKEYS.SESSION, {error: errorMessage});
+            Onyx.merge(ONYXKEYS.CREDENTIALS, {login: null});
+        }
+    });
+}
+
+/**
+ * Clears the Onyx store and redirects user to the sign in page
+ */
+function signOut() {
+    Timing.clearData();
+    redirectToSignIn();
+    if (!credentials || !credentials.login) {
+        return;
+    }
+
+    API.DeleteLogin({
+        partnerUserID: credentials.login,
+        partnerName: CONFIG.EXPENSIFY.PARTNER_NAME,
+        partnerPassword: CONFIG.EXPENSIFY.PARTNER_PASSWORD,
+        doNotRetry: true,
+    })
+        .catch(error => Onyx.merge(ONYXKEYS.SESSION, {error: error.message}));
+}
+
+/**
+ * Checks the API to see if an account exists for the given login
+ *
+ * @param {String} login
+ */
+function fetchAccountDetails(login) {
+    Onyx.merge(ONYXKEYS.SESSION, {error: ''});
+
+    API.GetAccountStatus({email: login})
+        .then((response) => {
+            if (response.jsonCode === 200) {
+                Onyx.merge(ONYXKEYS.CREDENTIALS, {login});
+                Onyx.merge(ONYXKEYS.ACCOUNT, {
+                    accountExists: response.accountExists,
+                    canAccessExpensifyCash: response.canAccessExpensifyCash,
+                    requiresTwoFactorAuth: response.requiresTwoFactorAuth,
+                });
+
+                if (!response.accountExists) {
+                    createAccount(login);
+                }
+            }
+        });
+}
+
+/**
+ * Sign the user into the application. This will first authenticate their account
+ * then it will create a temporary login for them which is used when re-authenticating
+ * after an authToken expires.
+ *
+ * @param {String} password
  * @param {String} [twoFactorAuthCode]
- * @param {String} [exitTo]
+ * @param {String} exitTo
```

Comment:
Required params should be before optional ones.

Reply:
I'll fix that in my next PR